consistent-sampling/src/main/java/io/opentelemetry/contrib/sampler/consistent56/ConsistentSampler.java

```diff
@@ -186,10 +186,20 @@ public final SamplingResult shouldSample(
     boolean isSampled;
     boolean isAdjustedCountCorrect;
     if (isValidThreshold(threshold)) {
-      long randomness = getRandomness(otelTraceState, traceId);
-      isSampled = threshold <= randomness;
       isAdjustedCountCorrect = intent.isAdjustedCountReliable();
-    } else { // DROP
+      // determine the randomness value to use
+      long randomness;
+      randomness = getRandomness(otelTraceState, traceId);
```

Comment:
The value is set in the `if` and the `else`. I think either don't default it here, or remove the `if` block by inverting and keeping the `else`.
 
```suggestion
      long randomness;
```